modules/apigee_edge_apidocs/tests/src/Functional/ApiDocsAdminTest.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/**
+ * Copyright 2019 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\Tests\apigee_edge_apidocs\Functional;
+
+use Drupal\file\Entity\File;
+use Drupal\Tests\BrowserTestBase;
+use Drupal\Tests\TestFileCreationTrait;
+
+/**
+ * Simple test to ensure that main page loads with module enabled.
+ *
+ * @group apigee_edge
+ * @group apigee_edge_apidocs
+ */
+class ApiDocsAdminTest extends BrowserTestBase {
+
+  use TestFileCreationTrait;
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = ['apigee_edge_apidocs', 'block', 'field_ui'];
+
+  /**
+   * A user with permission to administer site configuration.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $adminUser;
+
+  /**
+   * Set up menus and tasks in their regions.
+   *
+   * Since menus and tasks are now blocks, we're required to explicitly set them
+   * to regions.
+   */
+  protected function setupMenus() {
+    $this->drupalPlaceBlock('local_actions_block', ['region' => 'content']);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    // Add the system menu blocks to appropriate regions.
+    $this->setupMenus();
+
+    $this->adminUser = $this->drupalCreateUser([
+      'add apidoc entities',
```

Comment:
Granting entity type's admin permission should be enough if you would like create an admin user for API doc entities. Granular CRUDL permissions should not be granted because that could hide some possible bugs, ex.: What if the entity access check for entity admin permission is not implemented properly?.

Reply:
Actually, what would be better is dedicated, granular, test cases to all CRUDL related permission which could ensure that a user with "add apidoc entities" permission **only** has access to the add API documentation form, etc.

Reply:
But this would be an overkill for a simple module like this. This was one of the reasons why I suggested to remove custom permissions and access implemtations from the original #129 PR created by @cnovak and depend on the [entity permission handler](https://github.com/fago/entity/blob/8.x-1.x/src/EntityPermissionProvider.php) and [entity access checker](https://github.com/fago/entity/blob/8.x-1.x/src/EntityAccessControlHandler.php) provided by the Entity module, which also ships test coverage for these implementations.

https://github.com/fago/entity/blob/8.x-1.x/tests/src/Unit/EntityAccessControlHandlerTest.php
https://github.com/fago/entity/blob/8.x-1.x/tests/src/Unit/EntityPermissionProviderTest.php

Reply:
We had a follow up for that, would it be okay to follow that? I do 100% agree with you, the value of this module is not in providing custom access logic.